### ISSUE TITLE
Juttlespec should verify the number of warnings/errors raised

### ISSUE THREAD

**Issue by bkutil** [juttle/juttle#430]
Currently, the generated code verifies that the warnings/errors are present (and in a given order): 

`expect(res.warnings[0]).to.include(...)`.

However, if less than expected warnings are emitted, the tests pass. Ideally, juttlespec should also do something like

`expect(warnings.length).to.equal(emitted_warnings.length)`.

so that the tests fail if some warnings/errors are missing.

**Comment by dmajda** [comment]
> However, if less than expected warnings are emitted, the tests pass.

JuttleSpec tests indeed fail when some expected warnings/errors are missing, which can be demonstrated with the following example (produces 1 warning, expects 2):

```markdown
# Test spec

## Emits warnings

### Juttle

    emit -from :0: -limit 1 | put time = time + null

### Warnings

  * Invalid operand types for "+": date (1970-01-01T00:00:00.000Z) and null.
  * Invalid operand types for "+": date (1970-01-01T00:00:01.000Z) and null.
```

On the other hand, JuttleSpec tests pass when there are more than expected warnings/errors, which can be demonstrated with the following example (produces 2 warnings, expects 1):

```markdown
# Test spec

## Emits warnings

### Juttle

    emit -from :0: -limit 2 | put time = time + null

### Warnings

  * Invalid operand types for "+": date (1970-01-01T00:00:00.000Z) and null.
```

@bkutil Was the latter case what you really meant?

**Comment by bkutil** [comment]
@dmajda - yes, the latter case is correct - I mistakenly wrote 'less' instead of 'more' in the original description.

**Comment by dmajda** [comment]
OK. Marking as a bug as the intention was always to match the errors/warnings exactly.

**Comment by bkutil** [comment]
Fixed in #602
